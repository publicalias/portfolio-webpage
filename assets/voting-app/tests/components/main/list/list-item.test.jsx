"use strict";

//local imports

const ListItem = require("../../../../scripts/components/main/list/list-item");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { newUser } = require("schemas/master");
const { newPoll } = require("schemas/voting-app");
const { initTestSnapshot, reactTests } = require("test-helpers/react-tests");

//node modules

const { MemoryRouter } = require("react-router-dom");

//setup

beforeAll(reactTests.setup);

//list item

describe("list item", () => {

  const { testShallow } = testWrapper(ListItem, MemoryRouter);

  const testSnapshot = initTestSnapshot(testShallow);

  const data = {
    title: "Title A",
    id: "id-a"
  };

  const testVotes = (mag) => testSnapshot({}, {
    poll: newPoll({
      users: { voted: 10 ** mag },
      ...data
    })
  });

  it("should match snapshot (default)", () => testSnapshot({}, { poll: newPoll(data) }));

  it("should match snapshot (authenticated)", () => testSnapshot({ user: newUser() }, { poll: newPoll(data) }));

  it("should match snapshot (10^3 votes)", () => testVotes(3));

  it("should match snapshot (10^6 votes)", () => testVotes(6));

  it("should match snapshot (10^9 votes)", () => testVotes(9));

});
