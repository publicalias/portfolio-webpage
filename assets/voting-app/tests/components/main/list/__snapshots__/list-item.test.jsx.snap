// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list item should match snapshot (10^3 votes) 1`] = `
<tr>
  <td>
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": 0,
          "polls": Array [],
          "user": Object {},
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "list": true,
          "poll": Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "Title A",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 1000,
            },
          },
          "role": "hide",
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </td>
  <td
    className="u-cursor-pointer"
  >
    <Link
      to="/view?id=id-a"
    >
      <p>
        Title A
      </p>
      <p
        className="u-no-margin"
      >
        1K Votes — 31 December 1969
      </p>
    </Link>
  </td>
</tr>
`;

exports[`list item should match snapshot (10^6 votes) 1`] = `
<tr>
  <td>
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": 0,
          "polls": Array [],
          "user": Object {},
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "list": true,
          "poll": Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "Title A",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 1000000,
            },
          },
          "role": "hide",
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </td>
  <td
    className="u-cursor-pointer"
  >
    <Link
      to="/view?id=id-a"
    >
      <p>
        Title A
      </p>
      <p
        className="u-no-margin"
      >
        1M Votes — 31 December 1969
      </p>
    </Link>
  </td>
</tr>
`;

exports[`list item should match snapshot (10^9 votes) 1`] = `
<tr>
  <td>
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": 0,
          "polls": Array [],
          "user": Object {},
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "list": true,
          "poll": Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "Title A",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 1000000000,
            },
          },
          "role": "hide",
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </td>
  <td
    className="u-cursor-pointer"
  >
    <Link
      to="/view?id=id-a"
    >
      <p>
        Title A
      </p>
      <p
        className="u-no-margin"
      >
        ??? Votes — 31 December 1969
      </p>
    </Link>
  </td>
</tr>
`;

exports[`list item should match snapshot (authenticated) 1`] = `
<tr>
  <td>
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": 0,
          "polls": Array [],
          "user": Object {
            "auth": Object {
              "id": "",
              "provider": "",
            },
            "data": Object {
              "restricted": false,
            },
            "id": "",
            "name": "",
            "type": "auth",
          },
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "list": true,
          "poll": Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "Title A",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 0,
            },
          },
          "role": "hide",
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </td>
  <td>
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": 0,
          "polls": Array [],
          "user": Object {
            "auth": Object {
              "id": "",
              "provider": "",
            },
            "data": Object {
              "restricted": false,
            },
            "id": "",
            "name": "",
            "type": "auth",
          },
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "list": true,
          "poll": Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "Title A",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 0,
            },
          },
          "role": "flag",
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </td>
  <td
    className="u-cursor-pointer"
  >
    <Link
      to="/view?id=id-a"
    >
      <p>
        Title A
      </p>
      <p
        className="u-no-margin"
      >
        0 Votes — 31 December 1969
      </p>
    </Link>
  </td>
</tr>
`;

exports[`list item should match snapshot (default) 1`] = `
<tr>
  <td>
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": 0,
          "polls": Array [],
          "user": Object {},
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "list": true,
          "poll": Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "Title A",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 0,
            },
          },
          "role": "hide",
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </td>
  <td
    className="u-cursor-pointer"
  >
    <Link
      to="/view?id=id-a"
    >
      <p>
        Title A
      </p>
      <p
        className="u-no-margin"
      >
        0 Votes — 31 December 1969
      </p>
    </Link>
  </td>
</tr>
`;
